src/services/chain/messages/kava/evmutil/erc20Conversion.ts

```diff
@@ -0,0 +1,238 @@
+import Kava from '@kava-labs/javascript-sdk';
+import { InProgressTxDisplay } from '../../../../../components/InProgressTxDisplay';
+import {
+  ChainNames,
+  chainRegistry,
+  CosmosChainConfig,
+  EVMChainConfig,
+} from '../../../../../config/chainsRegistry';
+import {
+  ChainMessage,
+  ChainType,
+  OperationType,
+} from '../../../../../types/chain';
+import {
+  getCoinRecord,
+  getERC20Record,
+} from '../../../../../utils/chat/helpers';
+import {
+  SignatureTypes,
+  WalletStore,
+  WalletTypes,
+} from '../../../../../walletStore';
+import { ethers } from 'ethers';
+import { bech32 } from 'bech32';
+import { erc20ABI } from '../../../../../tools/erc20ABI';
+import { EIP712SignerParams } from '../../../../../eip712';
+import { walletStore } from '../../../../../stores';
+import { Message } from '@kava-labs/javascript-sdk/lib/types/Message';
+
+interface ERC20ConvertParams {
+  chainName: string;
+  amount: string;
+  denom: string;
+  direction: 'coinToERC20' | 'ERC20ToCoin';
+}
+
+export class ERC20ConversionMessage
+  implements ChainMessage<ERC20ConvertParams>
+{
+  name = 'erc20Convert';
+  description =
+    'Converts an ERC20 asset to a cosmos Coin or cosmos coin to an ERC20 asset';
+  chainType = ChainType.COSMOS;
+  operationType = OperationType.TRANSACTION;
+  walletMustMatchChainID = true;
+
+  needsWallet = [WalletTypes.METAMASK];
+
+  /**
+   * Parameter definitions for the message.
+   * Used for validation and OpenAI tool generation.
+   */
+  parameters = [
+    {
+      name: 'chainName',
+      type: 'string',
+      description: `name of the chain the user is interacting with, if not specified by the user assume ${ChainNames.KAVA_COSMOS}`,
+      enum: [...Object.keys(chainRegistry[ChainType.COSMOS])],
+      required: true,
+    },
+    {
+      name: 'amount',
+      type: 'string',
+      description: 'Amount to convert',
+      required: true,
+    },
+    {
+      name: 'denom',
+      type: 'string',
+      description: 'Token denomination',
+      required: true,
+    },
+    {
+      name: 'direction',
+      type: 'string',
+      description:
+        'direction of the conversion either from cosmos Coin to erc20 or erc20 to cosmos Coin',
+      enum: ['coinToERC20', 'ERC20ToCoin'],
+      required: true,
+    },
+  ];
+
+  inProgressComponent() {
+    return InProgressTxDisplay;
```

Comment:
I think we will want to show the connect wallet prompt here so that a user gets directed to metamask.

![Screenshot 2025-01-31 at 12 17 15 PM](https://github.com/user-attachments/assets/e38121f1-14bf-4d5b-8d67-f47759dccc91)

https://github.com/Kava-Labs/oros/blob/eba1154abb244f89e0d59b3abe84e52f7cfe28f6/src/services/chain/messages/evm/transfer.ts#L60